services/src/main/java/edu/unc/lib/dl/cdr/services/processing/UpdateMODSService.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright 2008 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.cdr.services.processing;
+
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.util.Arrays;
+
+import org.apache.abdera.Abdera;
+import org.apache.abdera.model.Document;
+import org.apache.abdera.model.Entry;
+import org.apache.abdera.parser.Parser;
+
+import edu.unc.lib.dl.acl.service.AccessControlService;
+import edu.unc.lib.dl.acl.util.AgentPrincipals;
+import edu.unc.lib.dl.acl.util.Permission;
+import edu.unc.lib.dl.fcrepo4.ContentObject;
+import edu.unc.lib.dl.fcrepo4.RepositoryObjectLoader;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.services.OperationsMessageSender;
+import edu.unc.lib.dl.update.AtomPubMetadataUIP;
+import edu.unc.lib.dl.update.MODSValidationUIPFilter;
+import edu.unc.lib.dl.update.UIPException;
+import edu.unc.lib.dl.update.UpdateOperation;
+
+/**
+ * Service that manages MODS description updates
+ *
+ * @author harring
+ *
+ */
+public class UpdateMODSService {
+
+    private AccessControlService aclService;
+    private RepositoryObjectLoader repoObjLoader;
+    private OperationsMessageSender operationsMessageSender;
+    private MODSValidationUIPFilter modsFilter;
```

Comment:
I would recommend not using this class since all of the UIP things are intended to be part of a pipeline and have a number of fedora 3 expectations in them.

It would actually be good to have a common MODS validator class, that we could use here and in the deposit pipeline.  Basically duplicate the deposit pipeline code.  Provide a premis logger, whatever agent should be used, and a mods document.

Reply:
The abdera stuff only shows up in the update pipeline because it was being pulled out of sword requests, which you shouldn't need to deal with here.